packages/server-wallet/src/models/funding.ts

```diff
@@ -47,13 +61,40 @@ export class Funding extends Model implements RequiredColumns {
       .first();
 
     if (!existing) {
-      return await Funding.query(knex).insert({channelId, amount, assetHolder});
+      return await Funding.query(knex).insert({channelId, amount, assetHolder, transferredOut: []});
     } else {
       return await Funding.query(knex)
-        .patch({channelId, amount, assetHolder})
+        .patch({amount})
         .where({channelId, assetHolder})
         .returning('*')
         .first();
     }
   }
+
+  static async updateTransferredOut(
+    knex: Knex,
+    channelId: Bytes32,
+    assetHolder: Address,
+    toAddress: Uint256,
+    amount: Address
+  ): Promise<Funding> {
+    return knex.transaction(async tx => {
+      const existing = await Funding.query(tx)
+        .where({channelId, assetHolder})
+        .first();
+
+      if (!existing) {
+        const errorMessage = `Expected for funding row to exists with channelId ${channelId}, assetHolder ${assetHolder}`;
+        logger.error(errorMessage, {channelId, assetHolder, toAddress, amount});
+        throw new Error(errorMessage);
```

Comment:
Alternative way to implement this: https://vincit.github.io/objection.js/api/query-builder/other-methods.html#throwifnotfound

Reply:
Excellent point: 07b806a33